modules/client/src/client-common/core.ts

```diff
@@ -13,13 +13,24 @@ import { GraphqlModule } from "./modules/graphql";
  * Provides the low level foundation so that subclasses have ready-made access to Web3, IPFS and GraphQL primitives
  */
 export abstract class ClientCore implements IClientCore {
-  public web3: IClientWeb3Core;
-  public ipfs: IClientIpfsCore;
-  public graphql: IClientGraphQLCore;
+  private privateWeb3: IClientWeb3Core;
+  private privateIpfs: IClientIpfsCore;
+  private privateGraphql: IClientGraphQLCore;
 
   constructor(context: Context) {
-    this.web3 = new Web3Module(context);
-    this.ipfs = new IPFSModule(context);
-    this.graphql = new GraphqlModule(context);
+    this.privateWeb3 = new Web3Module(context);
+    this.privateIpfs = new IPFSModule(context);
+    this.privateGraphql = new GraphqlModule(context);
+    Object.freeze(ClientCore.prototype);
```

Comment:
If you freeze `this` you cannot extend the class afterwards and this will crash
```ts
export class MyClass extends ClientCore {
//...
}
```

Reply:
Then we also need to use a `Map` here, because the private field can still be accessed and modified from the outside.

The getters don't add anything here

Reply:
I don't think so, because we freeze the objects that extend this class, and this class is not used elsewhere

Reply:
yes, we freeze the inner objects on their scope, but not on the root `client` scope. So I could come and do `client["privateWeb3"] = new MyMaliciousWeb3()` without a problem

Reply:
If the only thing exported is the getters, and the getters wrap 3 private maps, then the components cannot be directly accessed

Reply:
I believe it freezes everything, including the objects and functions in the ClientCore